### ISSUE TITLE
POS Add To Cart doesn't support varied line item properties

### ISSUE THREAD

**Issue by msadlows** [Shopify/shopify-app-bridge#28]
This bug relates to all versions of App Bridge, all POS devices, and all instances of App Bridge when interacting with the cart.

Currently, when adding to the cart in POS, app bridge allows us to specify the variant and quantity we want to add, amongst a few other things if it is a custom product. It however does not allow us to specify line item properties with that payload. Instead, we need to add line item properties based on the line item index after the fact. This creates an issue where you cannot add the same variant with different line item properties. 

### Example
An example to illustrate how this differs from the Online Store. You sell custom rings with engraving, and your customer wants to buy three rings for his daughters Jill, Kelly and Laura.  On the online store, you can effectively add the same variant to the cart three times with a line item property of "engraving" equal to the name.

In POS, there is no way to add three separate line items of the same variant, they simply combine and become a quantity of three. From there, you are unable to split that line item into multiples to apply different line item properties.

### Previous Suggestions
The answer I have received in the past is to simply pass multiple keys to the line item property value of that combined line item, such as "engraving1", "engraving2", and "engraving3". This doesn't work though, as it doesn't then match the online store orders. Downstream systems will therefore not work or require additional editing. 

The other answer was to modify the order after the fact. This isn't possible as the GQL doesn't support this currently and the initial order/create hook would go out before we could change the details. 

I see this as a major bug in App Bridge for POS and it has been something that has severally delayed our ability to develop solutions for POS that merchants are asking for constantly. We have built an app that requires this functionality to actually release, but we need this bug fixed before we can move forward.

### Proposed Solution
Three things would need to be done to support this:

1. Add the ability to send line item property key/value pairs when adding a variant line item to the POS cart through the APP::CART::ADD_LINE_ITEM action
2. Ensure that POS properly keeps line items separate when the line item properties are different (This may already function as is)

**Comment by ckeyes88** [comment]
This is having a pretty big impact on our Shopify POS integration. Can @andrewapperley or @iainmcampbell speak to the status of this issue?

**Comment by msadlows** [comment]
@ckeyes88 It's really interesting to see this as my conversation with POS product managers has seemingly suggested that this is on their radar and being looked at. I think the key word here is "near-term", which is ambiguous. Hopefully it gets some more attention, it is a major blocker for anyone trying to tie POS orders properly into those on the online store and other sales channels.

**Comment by ckeyes88** [comment]
@msadlows Good point. It's certainly true that things take time so I'm glad to hear that it's on their radar and hopefully near-term doesn't mean too far out.

**Comment by pablogiralt** [comment]
@msadlows @ckeyes88 I have added this issue to the new POS UI Extensions repo (see link above).  Maybe we are lucky and the team behind the extensions considers this to be as important as we know it is...
Would be great that you add your +1 there

**Comment by ckeyes88** [comment]
Hey @pablogiralt I will definitely do that! I can't believe they've ignored it for so long.
